pkg/operation/botanist/matchers/matcher.go

```diff
@@ -64,8 +64,8 @@ var (
 	// WebhookConstraintMatchersForLeases contains a list of lease API resources that can break
 	// leader election of essential control plane controllers.
 	WebhookConstraintMatchersForLeases = []WebhookConstraintMatcher{
-		{GVR: coordinationv1.SchemeGroupVersion.WithResource("leases"), NamespaceLabels: kubeSystemNamespaceLabels},
-		{GVR: coordinationv1beta1.SchemeGroupVersion.WithResource("leases"), NamespaceLabels: kubeSystemNamespaceLabels},
+		{GVR: coordinationv1.SchemeGroupVersion.WithResource("leases"), NamespaceLabels: kubeSystemNamespaceLabels, ObjectLabels: labels.Set{}},
```

Comment:
Allowing and object selector could be problematic later, if somebody intentionally tries to select kubernetes controlplane objects. Maybe put a comment about this, and why we have to use the ObjectLabels for now.